crates/astria-conductor/src/conductor.rs

```diff
@@ -319,6 +314,49 @@ impl Conductor {
     }
 }
 
+async fn get_executor(
+    cfg: &Config,
+    block_rx: UnboundedReceiver<crate::executor::ExecutorCommand>,
+    shutdown_rx: Receiver<()>,
+) -> eyre::Result<Executor> {
+    let hook = if cfg.enable_optimism {
+        let provider = Arc::new(
+            Provider::<Ws>::connect(cfg.ethereum_l1_url.clone())
```

Comment:
Does this have to be `clone()`d? Doesn't work with a `&str`?

Reply:
no :(